odl/discr/diff_ops.py

```diff
@@ -172,7 +195,7 @@ def adjoint(self):
         return -PartialDerivative(self.domain, self.axis,
                                   _ADJ_METHOD[self.method],
                                   _ADJ_PADDING[self.pad_mode],
-                                  self.pad_const, self.edge_order)
+                                  self.pad_const)
 
 
 class Gradient(PointwiseTensorFieldOperator):
```

Comment:
I gave it some thought but realized that it may have practical issues. The current implementation of `PointwiseTensorFieldOperator` probably prohibits this case. Maybe not for the gradient, which could map from `X` to `X ** len(indices)`, but for the divergence it would be harder since we cannot handle `X ** ndim --> X ** (ndim - len(indices))` in any nice way.

I guess this has to wait for proper tensors. In the meanwhile, we can always to
```python
BroadcastOperator(*(PartialDerivative(space, axis) for axis in indices))
```
for the gradient and
```python
ReductionOperator(*(PartialDerivative(space, axis) for axis in indices))
```
for the divergence.